packages/shared_preferences/shared_preferences_android/android/src/main/java/io/flutter/plugins/sharedpreferences/MethodCallHandlerImpl.java

```diff
@@ -109,6 +110,10 @@ public void onMethodCall(MethodCall call, MethodChannel.Result result) {
         case "getAll":
           result.success(getAllPrefs());
           return;
+        case "getAllWithPrefix":
+          String prefix = call.argument("prefix");
+          result.success(getAllPrefs(prefix));
+          return;
         case "remove":
           commitAsync(preferences.edit().remove(key), result);
           break;
```

Comment:
I can't comment a few lines down, but the implementation of `clear` is now incorrect on all platforms, because it's using the default prefix. You'll need a `clearAllWithPrefix` method too.

The flow of:
- set a custom prefix
- set some values
- make sure you can read them back
- clear
- make sure you can't read them any more

would be a great candidate for an integration test.